bzzeth/bzzeth.go

```diff
@@ -266,14 +269,28 @@ func (b *BzzEth) deliverAndStoreAll(ctx context.Context, req *request, headers [
 
 	// wait for all validations to get over and close the channels
 	err := wg.Wait()
-	if err != nil {
-		return err
+
+	// We want to store even if there is any validation error.
+	// since some headers may be valid in the batch.
+	// Store all the valid header chunks in one shot
+	storeErr := b.storeChunks(ctx, chunks)
+	if storeErr != nil {
+		return storeErr
 	}
 
+	// Pass on the error if any from the validation error group above storage
+	return err
+}
+
+func (b *BzzEth) storeChunks(ctx context.Context, chunks []chunk.Chunk) error {
 	// Store all the valid header chunks in one shot
 	results, err := b.netStore.Put(ctx, chunk.ModePutUpload, chunks...)
 	if err != nil {
-		for i := range results {
+		for i, flag := range results {
+			// if this chunk is stored successfully, dont report error for it
+			if !flag {
+				continue
+			}
```

Comment:
you still do this error checking on line 296 for every iteration. instead just put it before the loop.
and also we should not ignore other errors coming from the db

Reply:
Pushed the err outside the loop
Catch other errors also

Reply:
Done